src/components/base/BaseButton.vue

```diff
@@ -0,0 +1,279 @@
+    arrowType: {
+      type: String,
+      default: 'none',
+      validator(value) {
+        return ['none', 'right', 'left'].includes(value);
+      },
+    },
+  },
+  computed: {
+    componentType() {
+      if (this.$attrs.href) return 'a';
+      // TODO: switch line below with the following one after implementation of ROUTING
+      return 'button';
+      // return this.to !== null ? 'router-link' : 'button';
+    },
+    buttonClasses() {
+      return {
+        'base-button': true,
+        'base-button--large': this.size === 'large',
+        'base-button--regular': this.size === 'regular',
+        'base-button--medium': this.size === 'medium',
+        'base-button--small': this.size === 'small',
+        'base-button--filled-red': this.buttonType === 'filled-red',
+        'base-button--hollow-red': this.buttonType === 'hollow-red',
+        'base-button--hollow-red-faded': this.buttonType === 'hollow-red-faded',
+        'base-button--filled-dark': this.buttonType === 'filled-dark',
+        'base-button--hollow-dark': this.buttonType === 'hollow-dark',
+        'base-button--borderless-red': this.buttonType === 'borderless-red',
+        'base-button--right': this.arrowType === 'right',
+        'base-button--left': this.arrowType === 'left',
+      };
+    },
+    iconClasses() {
+      return {
+        'base-button__icon': true,
+        'base-button__icon--right': this.arrowType === 'right',
+        'base-button__icon--left': this.arrowType === 'left',
+      };
+    },
+  },
+});
+</script>
+
+<template>
+  <!-- TODO: switch line below with the following one after implementation of ROUTING -->
+  <component :is="componentType" :class="buttonClasses" @click="this.$emit('btnClick')">
+    <!-- <component :is="componentType" :class="buttonClasses" :to="to" @click="this.$emit('btnClick')"> -->
+    <ArrowLeftIcon v-if="arrowType === 'left'" :class="iconClasses" />
+    <slot>button_label</slot>
+    <ArrowRightIcon v-if="arrowType === 'right'" :class="iconClasses" />
+  </component>
+</template>
+
+<style lang="scss" scoped>
+.base-button {
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  gap: $padding-small;
+
+  width: fit-content;
+
+  font-family: 'Roboto Mono', monospace;
+  text-align: center;
+  line-height: 100%;
+  text-decoration: none;
+
+  border: $button-border-width solid;
+  border-radius: $border-radius-full;
+  outline: none;
+
+  user-select: none;
+  cursor: pointer;
+
+  transition: transform ease 75ms;
+
+  &:focus-visible,
+  &:active {
+    transform: translateY(0.2rem);
+  }
+
+  &--right {
+    &:focus-visible,
+    &:active {
+      transform: translateX(0.2rem);
+    }
+  }
+
+  &--left {
+    &:focus-visible,
+    &:active {
+      transform: translateX(-0.2rem);
+    }
+  }
+
+  &--filled-red {
+    color: $white;
+    background-color: $red-cherryred;
+    border-color: $red-cherryred;
+    font-weight: $font-weight-regular;
+
+    svg {
+      stroke: $white;
+    }
```

Comment:
Actually, if it's always that stroke has the color of the `color` property in the button, we could only declare it once and do:

```css
svg {
  stroke: currentColor;
}
```

This way, whatever color we set it will be applied to the svg's stroke 😄

Reply:
Amazing! I didn't know properties like that even exist 😃
That will save a lot of lines in the component styling